<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>Monophylizer</title>
    
    <link rel="icon" href="img/favicon.png" sizes="32x32">

    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/scrolling-nav.css" rel="stylesheet">
    
    <!-- Wizard CSS -->
    <link href="css/prettify.css" rel="stylesheet">    
	
    <!-- Widget CSS -->
    <link href="css/jasny-bootstrap.min.css" rel="stylesheet">
	
    <style type="text/css">
	.panel { text-align: left; margin-top: 1em }
	a.glyphicon-info-sign { text-decoration: none }
    </style>

    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
    <![endif]-->

</head>

<!-- The #page-top ID is part of the scrolling feature - the data-spy and data-target are part of the built-in Bootstrap scrollspy function -->

<body id="page-top" data-spy="scroll" data-target=".navbar-fixed-top">

    <!-- Navigation -->
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container">
            <div class="navbar-header page-scroll">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                    <span class="sr-only">Toggle navigation</span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand page-scroll" href="#page-top">Monophylizer</a>
            </div>

            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse navbar-ex1-collapse">
                <ul class="nav navbar-nav">
                    <!-- Hidden li included to remove active class from about link when scrolled up past about section -->
                    <li class="hidden">
                        <a class="page-scroll" href="#page-top">Monophylizer</a>
                    </li>
                    <li>
                        <a class="page-scroll" href="#faq">FAQ</a>
                    </li>                                                          
                    <!--li>
                        <a class="page-scroll" href="#about">About</a>
                    </li-->
                </ul>
            </div>
            <!-- /.navbar-collapse -->
        </div>
        <!-- /.container -->
    </nav>

    <!-- Intro Section -->
    <section id="intro" class="intro-section">
        <div class="container">
            <div class="row">
                <div class="col-lg-12">
                    <!--h1>Monophylize!</h1-->
                    <img src="img/logo.png" alt="Monophylize!" />
					<div id="rootwizard">
						<ul id="wizard">
							<li><a href="#tab1" data-toggle="tab">Upload tree</a></li>
							<li><a href="#tab2" data-toggle="tab">Name processing</a></li>
							<li><a href="#tab3" data-toggle="tab">Output formatting</a></li>
							<li><a href="#tab4" data-toggle="tab">Analyze!</a></li>
						</ul>
						<form 
							action="/cgi-bin/monophylizer.pl" 
							method="POST" 
							enctype="multipart/form-data">
							
							<!-- Sets flag so that service returns HTML, not TSV -->
							<input type="hidden" name="cgi" value="1" />
							
							<div class="tab-content">
								<div class="tab-pane" id="tab1">
									<div class="panel panel-default">
										<!--div class="panel-heading">
											<h3 class="panel-title">Upload tree file</h3>
										</div-->
										<div class="panel-body">
										  
											<!--input type="file" name="infile" /-->											
											<div class="fileinput fileinput-new input-group" data-provides="fileinput">
											  <div class="form-control" data-trigger="fileinput"><i class="glyphicon glyphicon-file fileinput-exists"></i> <span class="fileinput-filename"></span></div>
											  <span class="input-group-addon btn btn-default btn-file"><span class="fileinput-new">Select file</span><span class="fileinput-exists">Change</span><input type="file" name="infile"></span>
											  <a href="#" class="input-group-addon btn btn-default fileinput-exists" data-dismiss="fileinput">Remove</a>
											</div>

											<strong>File format:</strong>																																
											<select name="format">
												<option value="newick" selected="selected">newick</option>
												<option value="nexus">nexus</option>
												<option value="nexml">nexml</option>
												<option value="phyloxml">phyloxml</option>
											</select>
									
											<br /><br />
											<div class="alert alert-info" role="alert">
												<span class="glyphicon glyphicon-info-sign" aria-hidden="true" data-trigger="focus" data-toggle="popover"></span>	
												<a role="button" data-toggle="collapse" href="#accordion">												
												Click here for more about tree file formats than you'll ever want to know</a>
												<div class="panel-group collapse" id="accordion">
												  <div class="panel panel-default">
													<div class="panel-heading" role="tab" id="headingOne">
													  <h4 class="panel-title">
														<a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
														  Newick
														</a>
													  </h4>
													</div>
													<div id="collapseOne" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingOne">
													  <div class="panel-body">
														The simplest and most common file format for phylogenetic trees is the <a href="http://evolution.genetics.washington.edu/phylip/newicktree.html">Newick</a> (or New Hampshire) format. 
														Newick files are plain text files whose sole contents consist of parenthetical statements describing one or more tree shapes. You can verify this by opening your tree file in a 
														text editor such as NotePad on Windows, or TextEdit on Mac OSX: if you only see lots of parentheses with text inside it's probably a Newick file.
														Regrettably, the Newick "standard" is not followed too carefully by different programs. We have attempted to be permissive in what the monophylizer accepts but 
														there are some caveats to keep in mind:
														<ul>
															<li class="info"><strong>Newick files can contain multiple tree descriptions. The monophylizer only reads the first tree.</strong></li>	
															<li class="info"><strong>The monophylizer operates strictly on topology. Branch lengths are therefore optional: you can use either cladograms or phylograms.
															If you do provide branch lengths, they can be arbitrary length integers, floating point numbers, or use scientific notation.</strong></li>											
															<li class="info">The tips in your tree should have names that consist of the species name (<code><em>Genus species</em></code> or <code><em>Genus species subspecies</em></code>), <strong>and</strong>
															some kind of separator (such as the <code>|</code> symbol), 
															<strong>and then</strong> some kind of unique identifier such as a specimen or sequence identifier. In total, each name should therefore look something like <code>'Genus species|ID2347'</code>
															You can specify which record separator symbol you're using in the 'Tree reading' tab. You can use other symbols besides <code>|</code> as long as they don't have special meaning in Newick. 
															The following are therefore disallowed: <code>,'"[]();:_</code>
															</li>												
															<li class="info">BOLD has a tendency to break the standard, especially when you export trees with more metadata than strictly needed. If you export trees from BOLD make sure the trees contain
															species names and unique identifiers, but nothing else. Metadata such as names of collection localities or taxonomic authorities are quite likely to contain characters that
															have special meanings in Newick trees, yet BOLD does nothing to prevent their unsafe inclusion. For example, a place name such as <code>Côte d'Azur</code> is problematic, firstly 
															because the accent circonflexe might not transfer correctly (this has to do with <a href="https://en.wikipedia.org/wiki/Character_encoding">character encoding</a>) and
															secondly because the apostrophe is interpreted as an opening quote for which a file reader that follows the standard will expect a closing quote. Likewise, any text that
															is <code>[inside square brackets]</code> is interpreted in a special way: file readers that follow the standard assume that such text is a comment that should normally be stripped
															out of the tree description (some programs insert special data in these comments, though this is "non standard"). When BOLD inserts square brackets into your tree 
															description this can break the tree reading, especially when there is an opening bracket but no closing one, something that we've actually seen "in the wild".</li>
															<li class="info">Whitespace is normally stripped out of Newick trees before they are read. If a name in a tree needs to have spaces in them, such as the space between the genus name
															and the specific epithet, then this is done either by using an underscore symbol <code>_</code> instead of a space, or by putting the entire name in quotes. This allows 
															certain programs to insert spaces in tree descriptions as indentation, to show the "depth" of the taxon in the tree description. When such trees are read by a standard compliant
															tree reader, all the indentation spaces are stripped, as they should. However, BOLD inserts spaces between genus names and specific epithets without quoting the names. Any 
															strictly valid tree reader is going to remove those spaces as well, which results in concatenation of the genus and species names. We provide a workaround for this (incorrect) 
															syntax variant in the 'Tree reading' tab, but it is up to you to know the meaning of whitespace because a tree reader can't guess this automatically.																									
															</li>
														</ul>
														Here are examples of usable Newick files:
														<ul>
															<li class="info">Newick files with quoted whitespace:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_quoted/Adelidae_outgroup_rerooted.nwk">Adelidae_outgroup_rerooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_quoted/Eriocraniidae_outgroup_rerooted.nwk">Eriocraniidae_outgroup_rerooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_quoted/Nepticulidae_outgroup_rerooted.nwk">Nepticulidae_outgroup_rerooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_quoted/NJ_Argyresthiidae.nwk">NJ_Argyresthiidae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_quoted/NJ_Coleophoridae.nwk">NJ_Coleophoridae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_quoted/Psychidae_outgroup_rerooted.nwk">Psychidae_outgroup_rerooted.nwk</a></li>														
																</ul>
															</li>
															<li class="info">Newick files with underscored whitespace:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_escaped/ML_Coleophoridae.nwk">ML_Coleophoridae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_escaped/ML_Argyresthiidae.nwk">ML_Argyresthiidae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_escaped/RAxML_bipartitions.Pterophoridae_rooted.nwk">RAxML_bipartitions.Pterophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_escaped/RAxML_bipartitions.Coleophoridae_rooted.nwk">RAxML_bipartitions.Coleophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_escaped/RAxML_bipartitions.Gracillariidae_rooted.nwk">RAxML_bipartitions.Gracillariidae_rooted.nwk</a></li>
																</ul>
															</li>
															<li class="info"><strong>Technically invalid</strong> whitespace usage as exported by BOLD, allowed by the monophylizer (if 'Keep whitespace in Newick species names' is checked):
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_unescaped_invalid/ML_Coleophoridae.nwk">ML_Coleophoridae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_unescaped_invalid/ML_Argyresthiidae.nwk">ML_Argyresthiidae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_unescaped_invalid/RAxML_bipartitions.Pterophoridae_rooted.nwk">RAxML_bipartitions.Pterophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_unescaped_invalid/RAxML_bipartitions.Coleophoridae_rooted.nwk">RAxML_bipartitions.Coleophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_unescaped_invalid/RAxML_bipartitions.Gracillariidae_rooted.nwk">RAxML_bipartitions.Gracillariidae_rooted.nwk</a></li>
																</ul>
															</li>
															<li class="info">Indented Newick files with underscored whitespace. These are technically <strong>valid</strong>, but the monophylizer only accepts these
															if 'Keep whitespace in Newick species names' is <strong>unchecked</strong>. This is so that the common case of BOLD trees as input can be processed with default settings:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_indented/ML_Coleophoridae.nwk">ML_Coleophoridae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_indented/ML_Argyresthiidae.nwk">ML_Argyresthiidae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_indented/RAxML_bipartitions.Pterophoridae_rooted.nwk">RAxML_bipartitions.Pterophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_indented/RAxML_bipartitions.Coleophoridae_rooted.nwk">RAxML_bipartitions.Coleophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_indented/RAxML_bipartitions.Gracillariidae_rooted.nwk">RAxML_bipartitions.Gracillariidae_rooted.nwk</a></li>
																</ul>
															</li>															
															<li class="info">Newick cladograms. All syntax rules about quoting and whitespace apply, the only difference is the absence of branch lengths:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_cladograms/ML_Coleophoridae.nwk">ML_Coleophoridae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_cladograms/ML_Argyresthiidae.nwk">ML_Argyresthiidae.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_cladograms/RAxML_bipartitions.Pterophoridae_rooted.nwk">RAxML_bipartitions.Pterophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_cladograms/RAxML_bipartitions.Coleophoridae_rooted.nwk">RAxML_bipartitions.Coleophoridae_rooted.nwk</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/newick_cladograms/RAxML_bipartitions.Gracillariidae_rooted.nwk">RAxML_bipartitions.Gracillariidae_rooted.nwk</a></li>
																</ul>
															</li>															
														</ul>
													  </div>
													</div>
												  </div>
												  <div class="panel panel-default">
													<div class="panel-heading" role="tab" id="headingTwo">
													  <h4 class="panel-title">
														<a class="collapsed" role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseTwo" aria-expanded="false" aria-controls="collapseTwo">
														  Nexus
														</a>
													  </h4>
													</div>
													<div id="collapseTwo" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingTwo">
													  <div class="panel-body">
														The <a href="https://dx.doi.org/10.1093%2Fsysbio%2F46.4.590">Nexus standard</a> is a more complex, plain text-based file format for phylogenetic data. Nexus files can contain other 
														data besides trees, such as sequence alignments or other types of character state matrices, distance matrices, and so on. Many commonly-used programs read and/or write Nexus data. These
														include FigTree, Mesquite, TreeAnnotator, MrBayes, MacClade, PAUP*, etc. Nexus files often - but by no means always - have the extension <code>*.nex</code> or <code>*.nxs</code>. You can recognize
														whether your file is a Nexus file by opening it in a text editor such as NotePad on Windows or TextEdit on Mac OSX. If the very first word of the file is <code>#nexus</code>, either in
														lower or in uppper case, then the file is most likely a Nexus file. Regrettably, the Nexus "standard" is not followed too carefully by different programs. We have attempted to be permissive 
														in what the monophylizer accepts but there are some caveats to keep in mind:
														<ul>
															<li class="info"><strong>Nexus files can contain multiple tree descriptions. The monophylizer only reads the first tree.</strong></li>	
															<li class="info"><strong>The monophylizer operates strictly on topology. Branch lengths are therefore optional: you can use either cladograms or phylograms.
															If you do provide branch lengths, they can be arbitrary length integers, floating point numbers, or use scientific notation.</strong></li>											
															<li class="info"><strong>Nexus files can contain data "blocks" besides trees. The monophylizer ignores all non-tree blocks.</strong></li>
															<li class="info">Tree descriptions in Nexus files follow the same syntax as Newick. However, in some cases the taxon names are replaced with integers that correspond with a numbered list
															of taxon names, the so-called "translation table". This is done to save space and cut down on redundant names in files that contain many trees for the same
															taxa (for example in Bayesian analyses). The monophylizer accepts either usage: names embedded inside the tree statements or a separate translation table are both fine.</li>
															<li class="info">The tips in your tree should have names that consist of the species name (<code><em>Genus species</em></code> or <code><em>Genus species subspecies</em></code>), <strong>and</strong>
															some kind of separator (such as the <code>|</code> symbol), 
															<strong>and then</strong> some kind of unique identifier such as a specimen or sequence identifier. In total, each name should therefore look something like <code>'Genus species|ID2347'</code>
															You can specify which record separator symbol you're using in the 'Tree reading' tab. You can use other symbols besides <code>|</code> as long as they don't have special meaning in Nexus. 
															The following are therefore disallowed: <code>,'"[]();:_</code>
															</li>												
															<li class="info">BOLD has a tendency to break the standard, especially when you export trees with more metadata than strictly needed. If you export trees from BOLD make sure the trees contain
															species names and unique identifiers, but nothing else. Metadata such as names of collection localities or taxonomic authorities are quite likely to contain characters that
															have special meanings in Nexus trees, yet BOLD does nothing to prevent their unsafe inclusion. For example, a place name such as <code>Côte d'Azur</code> is problematic, firstly 
															because the accent circonflexe might not transfer correctly (this has to do with <a href="https://en.wikipedia.org/wiki/Character_encoding">character encoding</a>) and
															secondly because the apostrophe is interpreted as an opening quote for which a file reader that follows the standard will expect a closing quote. Likewise, any text that
															is <code>[inside square brackets]</code> is interpreted in a special way: file readers that follow the standard assume that such text is a comment that should normally be stripped
															out of the tree description (some programs insert special data in these comments, though this is "non standard"). When BOLD inserts square brackets into your tree 
															description this can break the tree reading, especially when there is an opening bracket but no closing one, which we've seen "in the wild".</li>													
														</ul>
														Here are examples of usable Nexus files:
														<ul>
															<li class="info">Nexus files with underscored whitespace:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped/ML_Coleophoridae.nex">ML_Coleophoridae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped/ML_Argyresthiidae.nex">ML_Argyresthiidae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped/RAxML_bipartitions.Pterophoridae_rooted.nex">RAxML_bipartitions.Pterophoridae_rooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped/RAxML_bipartitions.Coleophoridae_rooted.nex">RAxML_bipartitions.Coleophoridae_rooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped/RAxML_bipartitions.Gracillariidae_rooted.nex">RAxML_bipartitions.Gracillariidae_rooted.nex</a></li>
																</ul>
															</li>
															<li class="info">Nexus files without a translation table:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped_no_translate/ML_Coleophoridae.nex">ML_Coleophoridae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped_no_translate/ML_Argyresthiidae.nex">ML_Argyresthiidae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped_no_translate/RAxML_bipartitions.Pterophoridae_rooted.nex">RAxML_bipartitions.Pterophoridae_rooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped_no_translate/RAxML_bipartitions.Coleophoridae_rooted.nex">RAxML_bipartitions.Coleophoridae_rooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_escaped_no_translate/RAxML_bipartitions.Gracillariidae_rooted.nex">RAxML_bipartitions.Gracillariidae_rooted.nex</a></li>
																</ul>
															</li>
															<li class="info">Nexus files with quoted whitespace:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_quoted/Adelidae_outgroup_rerooted.nex">Adelidae_outgroup_rerooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_quoted/Eriocraniidae_outgroup_rerooted.nex">Eriocraniidae_outgroup_rerooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_quoted/Nepticulidae_outgroup_rerooted.nex">Nepticulidae_outgroup_rerooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_quoted/NJ_Argyresthiidae.nex">NJ_Argyresthiidae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_quoted/NJ_Coleophoridae.nex">NJ_Coleophoridae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_quoted/Psychidae_outgroup_rerooted.nex">Psychidae_outgroup_rerooted.nex</a></li>														
																</ul>
															</li>													
															<li class="info">Nexus cladograms:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_cladograms/ML_Coleophoridae.nex">ML_Coleophoridae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_cladograms/ML_Argyresthiidae.nex">ML_Argyresthiidae.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_cladograms/RAxML_bipartitions.Pterophoridae_rooted.nex">RAxML_bipartitions.Pterophoridae_rooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_cladograms/RAxML_bipartitions.Coleophoridae_rooted.nex">RAxML_bipartitions.Coleophoridae_rooted.nex</a></li>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexus_cladograms/RAxML_bipartitions.Gracillariidae_rooted.nex">RAxML_bipartitions.Gracillariidae_rooted.nex</a></li>
																</ul>
															</li>															
														</ul>
													  </div>
													</div>
												  </div>
												  <div class="panel panel-default">
													<div class="panel-heading" role="tab" id="headingThree">
													  <h4 class="panel-title">
														<a class="collapsed" role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseThree" aria-expanded="false" aria-controls="collapseThree">
														  XML-based formats
														</a>
													  </h4>
													</div>
													<div id="collapseThree" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingThree">
													  <div class="panel-body">
														Many of the ambiguities and different dialects in plain text-based file formats are due to differences in the way whitespace is treated, the way data is broken up into "words" or tokens.
														Extensible markup language (XML), and the generic parsers that are available for it, go some way to removing these tokenization issues. In addition, because formal grammars can be
														defined and published, some other ambiguities (such as number formats) can be resolved as well. Two such grammars have been developed for phylogenetics:
														<a href="http://phyloxml.org">phyloXML</a> and <a href="http://nexml.org">NeXML</a>. The former was developed with use cases in comparative genomics (such as orthology 
														assessment) in mind, the latter was developed to represent the block-like data structure of Nexus files as XML. You can recognize these files by opening them in a text editor such as 
														NotePad on Windows or TextEdit on Mac OSX. In both cases, as for all XML documents, they are likely to start with a in instruction such as the following:
														<code>&lt;?xml version="1.0" encoding="UTF-8"?&gt;</code>, which is then followed, respectively, by a statement that begins with <code>&lt;phyloxml</code> for phyloXML, or
														with <code>&lt;nex:nexml</code> for NeXML. Assuming your files are valid (which can be verified using the respective grammars of the standards) there are only a few caveats to keep
														in mind when using these files in the monophylizer:
														<ul>
															<li class="info"><strong>Both phyloXML and NeXML files can contain multiple tree descriptions. The monophylizer only reads the first tree.</strong></li>	
															<li class="info"><strong>The monophylizer operates strictly on topology. Branch lengths are therefore optional: you can use either cladograms or phylograms.</strong></li>											
															<li class="info">The tips in your tree should have names that consist of the species name (<code><em>Genus species</em></code> or <code><em>Genus species subspecies</em></code>), <strong>and</strong>
															some kind of separator (such as the <code>|</code> symbol), 
															<strong>and then</strong> some kind of unique identifier such as a specimen or sequence identifier. In total, each name should therefore look something like <code>'Genus species|ID2347'</code>
															You can specify which record separator symbol you're using in the 'Tree reading' tab. You can use other symbols besides <code>|</code> as long as they don't have special meaning in Newick. 
															The following are therefore disallowed: <code>,'"[]();:_</code>
															</li>															
														</ul>
														Here are examples of usable XML-based files:
														<ul>
															<li class="info">Phylograms in phyloXML:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/Adelidae_outgroup_rerooted.phyloxml">Adelidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/Eriocraniidae_outgroup_rerooted.phyloxml">Eriocraniidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/ML_Argyresthiidae.phyloxml">ML_Argyresthiidae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/ML_Coleophoridae.phyloxml">ML_Coleophoridae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/NJ_Argyresthiidae.phyloxml">NJ_Argyresthiidae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/NJ_Coleophoridae.phyloxml">NJ_Coleophoridae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/Nepticulidae_outgroup_rerooted.phyloxml">Nepticulidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/Psychidae_outgroup_rerooted.phyloxml">Psychidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/RAxML_bipartitions.Coleophoridae_rooted.phyloxml">RAxML_bipartitions.Coleophoridae_rooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/RAxML_bipartitions.Gracillariidae_rooted.phyloxml">RAxML_bipartitions.Gracillariidae_rooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml/RAxML_bipartitions.Pterophoridae_rooted.phyloxml">RAxML_bipartitions.Pterophoridae_rooted.phyloxml</a>																																
																</ul>
															</li>
															<li class="info">Cladograms in phyloXML:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/Adelidae_outgroup_rerooted.phyloxml">Adelidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/Eriocraniidae_outgroup_rerooted.phyloxml">Eriocraniidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/ML_Argyresthiidae.phyloxml">ML_Argyresthiidae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/ML_Coleophoridae.phyloxml">ML_Coleophoridae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/NJ_Argyresthiidae.phyloxml">NJ_Argyresthiidae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/NJ_Coleophoridae.phyloxml">NJ_Coleophoridae.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/Nepticulidae_outgroup_rerooted.phyloxml">Nepticulidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/Psychidae_outgroup_rerooted.phyloxml">Psychidae_outgroup_rerooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/RAxML_bipartitions.Coleophoridae_rooted.phyloxml">RAxML_bipartitions.Coleophoridae_rooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/RAxML_bipartitions.Gracillariidae_rooted.phyloxml">RAxML_bipartitions.Gracillariidae_rooted.phyloxml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/phyloxml_cladograms/RAxML_bipartitions.Pterophoridae_rooted.phyloxml">RAxML_bipartitions.Pterophoridae_rooted.phyloxml</a>																
																
																</ul>
															</li>
															<li class="info">Phylograms in NeXML:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/Adelidae_outgroup_rerooted.xml">Adelidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/Eriocraniidae_outgroup_rerooted.xml">Eriocraniidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/ML_Argyresthiidae.xml">ML_Argyresthiidae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/ML_Coleophoridae.xml">ML_Coleophoridae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/NJ_Argyresthiidae.xml">NJ_Argyresthiidae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/NJ_Coleophoridae.xml">NJ_Coleophoridae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/Nepticulidae_outgroup_rerooted.xml">Nepticulidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/Psychidae_outgroup_rerooted.xml">Psychidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/RAxML_bipartitions.Coleophoridae_rooted.xml">RAxML_bipartitions.Coleophoridae_rooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/RAxML_bipartitions.Gracillariidae_rooted.xml">RAxML_bipartitions.Gracillariidae_rooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml/RAxML_bipartitions.Pterophoridae_rooted.xml">RAxML_bipartitions.Pterophoridae_rooted.xml</a>
																</ul>
															</li>
															<li class="info">Cladograms in NeXML:
																<ul>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/Adelidae_outgroup_rerooted.xml">Adelidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/Eriocraniidae_outgroup_rerooted.xml">Eriocraniidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/ML_Argyresthiidae.xml">ML_Argyresthiidae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/ML_Coleophoridae.xml">ML_Coleophoridae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/NJ_Argyresthiidae.xml">NJ_Argyresthiidae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/NJ_Coleophoridae.xml">NJ_Coleophoridae.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/Nepticulidae_outgroup_rerooted.xml">Nepticulidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/Psychidae_outgroup_rerooted.xml">Psychidae_outgroup_rerooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/RAxML_bipartitions.Coleophoridae_rooted.xml">RAxML_bipartitions.Coleophoridae_rooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/RAxML_bipartitions.Gracillariidae_rooted.xml">RAxML_bipartitions.Gracillariidae_rooted.xml</a>
																	<li class="info"><a target="_blank" href="https://raw.githubusercontent.com/naturalis/monophylizer/master/data/nexml_cladograms/RAxML_bipartitions.Pterophoridae_rooted.xml">RAxML_bipartitions.Pterophoridae_rooted.xml</a>																
																</ul>
															</li>														
														</ul>
													  </div>
													</div>
												  </div>
												</div>	
											</div>																	
										</div>
									</div>
								</div>								
								<div class="tab-pane" id="tab2">

									<div class="panel panel-default">
										<!--div class="panel-heading">
											<h3 class="panel-title">Name processing</h3>
										  </div-->
										  <div class="panel-body">
										  
											<div class="alert alert-info" role="alert">
											  	The following options configure how tip names are processed. The monophylizer operates on trees where every tip must have both a species
											  	name (a binomial or trinomial) and a unique identifier, which are demarcated using a record separator. When you export trees
											  	from BOLD this separator is by default the '|' symbol, so you can leave this setting as is when your trees come from BOLD. If you 
											  	generate your trees in some other way make sure you update the settings here accordingly.
											</div>
										  
											<strong>Record separator:</strong>
											<input 
												type="text" 
												name="separator" 
												value="|" />
											<a class="glyphicon glyphicon-info-sign" href="#" aria-hidden="true" data-toggle="popover" data-trigger="focus" data-content="The tip names in BOLD trees consist
												of a taxon name followed by metadata fields (specimen and sequence identifiers, localities, etc.).
												By default, the separator between these fields is the '|'
												symbol."></a>	
											<br />										  
										  											
											<br />
										  
											<input 
												type="checkbox" 
												checked="checked" 
												name="whitespace" />
											<strong>Keep whitespace in Newick species names</strong>	
											<a class="glyphicon glyphicon-info-sign" href="#" aria-hidden="true" data-toggle="popover" data-trigger="focus" data-content="In Newick trees, whitespace in species 
												names is usually escaped, either by replacing it with underscores 
												('_') or by putting the entire species name in quotes.
												This allows some programs to generate pretty indentation to
												show the 'depth' of a taxon within the tree string. BOLD does
												not follow this convention: whitespace occurs in unescaped
												form only to delimit words, so normally
												this needs to be kept in BOLD trees."></a>
											<br /><br />
													
											<input 
												type="checkbox" 
												name="trinomials" />                            
											<strong>Split subspecies</strong>												
											<a class="glyphicon glyphicon-info-sign" href="#" aria-hidden="true" data-toggle="popover" data-trigger="focus" data-content="By default, this service ignores all parts of a taxonomic name following
												the first two words. This means 
												that subspecies ('Foo bar vulgaris') are lumped with the 
												containing species ('Foo bar') and that polyphyly or paraphyly 
												is therefore not assessed separately for subspecies. If you check this box,
												subspecies will be split and assessed separately."></a>																																																												
										</div>
									</div>

								</div>
								<div class="tab-pane" id="tab3">
									<div class="panel panel-default">
										<!--div class="panel-heading">
											<h3 class="panel-title">Output</h3>
										  </div-->
										  <div class="panel-body">
										  
											<!--div class="alert alert-info" role="alert">
												<strong>Note!</strong>
											  	The following options deal with how to represent the output.
											</div-->										  
										  
											<strong>Write output as TSV:</strong>                           
											<input type="checkbox" name="astsv" />
											<a class="glyphicon glyphicon-info-sign" href="#" aria-hidden="true" data-toggle="popover" data-trigger="focus" data-content="By default, this web service produces its output as a 
												a web page (i.e. a marked up HTML table). By checking this
												box, the service instead produces a tab-separated 
												spreadsheet, which can be imported directly into a variety
												of programs, such as R."></a>	
												<br /><br />
												
											<strong>Additional metadata:</strong>
											<a class="glyphicon glyphicon-info-sign" href="#" aria-hidden="true" data-toggle="popover" data-trigger="focus" data-content="Here you can upload
											a tab-separated spreadsheet file whose fields will be combined with those produced by the algorithm. You would use this
											if you want to include more data about species (not individual specimens) in the output, so the records in this file
											and in the output are matched by taxon names."></a>
											
											<!--input type="file" name="infile" /-->											
											<div class="fileinput fileinput-new input-group" data-provides="fileinput">
											  <div class="form-control" data-trigger="fileinput"><i class="glyphicon glyphicon-file fileinput-exists"></i> <span class="fileinput-filename"></span></div>
											  <span class="input-group-addon btn btn-default btn-file"><span class="fileinput-new">Select file</span><span class="fileinput-exists">Change</span><input type="file" name="..."></span>
											  <a href="#" class="input-group-addon btn btn-default fileinput-exists" data-dismiss="fileinput">Remove</a>
											</div>
																						
										</div>
									</div>								
								</div>
								<div class="tab-pane" id="tab4">
									<div class="panel panel-default">
										<!--div class="panel-heading">
											<h3 class="panel-title">Analyze!</h3>
										  </div-->
										  <div class="panel-body">
											<input type="submit" value="Go" class="btn" />																						
										</div>
									</div>								
								</div>
								<ul class="pager wizard">
									<li class="previous first" style="display:none;"><a href="#">First</a></li>
									<li class="previous"><a href="#">Previous</a></li>
									<li class="next last" style="display:none;"><a href="#">Last</a></li>
									<li class="next"><a href="#">Next</a></li>
								</ul>
							</div>	
						</form>
					</div>
                </div>
            </div>
        </div>
    </section>  

    <!-- FAQ Section -->
    <section id="faq" class="faq-section">
        <div class="container">
            <div class="row">
                <div class="col-lg-12">
                    <h1>FAQ</h1>
                    <dl>
                    	<dt>How does the program determine what's para- or polyphyletic?</dt>
                    	<dd>
                    		The steps of the algorithm are described in the manuscript
                    		(see: 'How to cite') of which this service is a part. In short:
                    		<ol>
                    			<li>Extract all distinct taxon names from the tree</li>
                    			<li>Label all interior nodes with a pre- and a post-order
                    			index.</li>
                    			<li>For each taxon:
                    			<ol type="i"> 
                    				<li>Collect all tips that belong to it</li>
                    				<li>Find the MRCA for the collected tips</li>
                    				<li>Collect all descendants of the MRCA. If 
                    				this set is identical to the set of step 
                    				<code>i.</code> then the taxon is monophyletic
                    				and the analysis moves on to the next taxon.</li>
                    				<li>Collect all that subtend tips from the focal 
                    				taxon as well as at least one other taxon and 
                    				sort these their post-order index.</li>
                    				<li>Group the collected, sorted nodes 
                    				into distinct root-to-tip paths. Internal nodes 
                    				that are nested in each other are identified 
                    				(and collected in the same group) by checking that 
                    				the pre-order index of the focal node is larger, 
                    				and the post-order index of the focal node is smaller 
                    				than that of the next node. If there is more than 
                    				one distinct root-to-tip path (i.e., group), the 
                    				taxon is considered polyphyletic, otherwise 
                    				paraphyletic.</li>
                    				<li>For each first (i.e. most recent) node in each 
                    				group, collect all subtended species. The union of 
                    				these sets across groups forms the set of entangled 
                    				species.</li>
                    			</ol>
                    			</li>
                    		</ol>
                    	</dd>
                    	<dt>How do I interpret the results?</dt>
                    	<dd>
                    		The output of the algorithm is presented in tabular form. Each row
                    		represent one taxon from the tree. The 'assessment' column shows
                    		whether that taxon is mono-, para- or polyphyletic. The 'tanglees'
                    		column shows with which other taxa, if any, the focal taxon is entangled.
                    	</dd>
                    	<dt>How do I export the results?</dt>
                    	<dd>
                    		You can copy and paste the results from the browser window into
                    		a spreadsheet program. Easier still would be to check the 'TSV'
                    		box in the 'output' tab. The results will then be written as
                    		tab-separated data that can be imported directly into spreadsheet
                    		programs, R, etc.
                    	</dd>
                    	<dt>How is this algorithm implemented?</dt>
                    	<dd>
                    		The algorithm is written in the Perl programming language 
                    		and uses the 
                    		<a href="http://search.cpan.org/dist/Bio-Phylo">Bio::Phylo</a>
                    		libraries to read the input data.
                    	</dd>
                    	<dt>Can I use this "off-line"?</dt>
                    	<dd>
                    		Yes. You can run <a href="https://raw.githubusercontent.com/naturalis/monophylizer/master/script/monophylizer.pl">the script</a>
                    		locally. Consult the embedded Perl-Doc documentation in the
                    		script or run <code>perl monophylizer.pl --help</code> for
                    		more info.
                    	</dd>
                    	<dt>Can I use this through an API?</dt>
                    	<dd>
                    		Yes. The data and parameters are uploaded through an 
                    		<code>HTTP POST</code> request with <code>multipart/form</code> 
                    		data and the results are in the response, so this
                    		service functions as a RESTful web service. You will need
                    		to provide the following parameters:
                    		<ul>
                    			<li><code>infile</code>, which is a file upload</li>
                    			<li><code>format</code>, input file format, one of:
                    			<code>newick, nexus, nexml, phyloxml</code></li>
                    			<li><code>separator</code>, the character that separates
                    			the taxon name from its identifier. Default is |</li>
                    			<li><code>trinomials</code>, which is an optional argument
                    			that, when given any value other than '0', indicates that
                    			subspecific epithets need to be parsed</li>
                    			<li><code>astsv</code>, which is an optional argument
                    			that, when given any value other than '0', indicates that
                    			the output must be written as tab-separated data</li>
                    			<li><code>metadata</code>, which is an optional file
                    			upload with additional tab-separated data to join with
                    			the taxa in the output</li>
                    		</ul>
                    	</dd>
                    	<dt>How is this code licensed?</dt>
                    	<dd>
                    		The analysis script proper is licensed under the 
                    		<a href="https://raw.githubusercontent.com/naturalis/monophylizer/master/LICENSE">Apache License</a>, 
                    		which is very permissive for most forms of re-use.
                    	</dd>
                    	<dt>Who wrote this code?</dt>
                    	<dd>
                    		The analysis code was written by 
                    		<a href="http://rutgervos.blogspot.com">Rutger Vos</a>.
                    	</dd>
                    	<dt>Where is this hosted?</dt>
                    	<dd>
                    		The web service is hosted at <a href="http://www.naturalis.nl">
                    		Naturalis Biodiversity Center
                    		</a>, the source code in 
                    		<a href="https://github.com/naturalis/monophylizer">a GitHub repository</a>.
                    	</dd>
                    	<dt>How to cite?</dt>
                    	<dd>
                    		This service is part of a publication. If you use this service
                    		in your research, please cite the publication:<br />
                    		<blockquote>
                    		<strong>Mutanen, M.</strong> <em>et al.</em> 2016. Species-Level 
                    		Poly- and Paraphyly in DNA barcode Gene Trees: Strong Operational 
                    		Bias in European Lepidoptera. <em>Systematic Biology</em>
                    		</blockquote>
                    	</dd>
                    	<dt>How to format taxonomic names?</dt>
                    	<dd>
                    		Taxonomic names are read without any intelligence:
                    		the expectation is that names consist of the genus,
                    		the specific epithet, and, optionally, the subspecific
                    		epithet, followed by an identifier. Anything else,
                    		such as 'sp.', 'cv.', and so on are going to cause
                    		problems and will have to be avoided for this analysis.
                    	</dd>
                    	<dt>I am getting error messages?</dt>
                    	<dd>
                    		This is almost certainly because your input file is
                    		somehow syntactically invalid. On the 'upload' tab
                    		there is a link to extensive documentation, with
                    		example files, to explain what input file formats
                    		are accepted. If you can't figure out what's wrong
                    		with your file, try opening it in 
                    		<a href="http://mesquiteproject.org">Mesquite</a>
                    		and exporting it as NEXUS: its intepretation of the
                    		NEXUS standard is readily understood by this service,
                    		and Mesquite might give you useful feedback about
                    		your file.
                    	</dd>
                    </dl>
                </div>
            </div>
        </div>
    </section>

    <!-- About Section -->
    <!--section id="about" class="about-section">
        <div class="container">
            <div class="row">
                <div class="col-lg-12">
                    <h1>About</h1>
                </div>
            </div>
        </div>
    </section-->

    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>

    <!-- Scrolling Nav JavaScript -->
    <script src="js/jquery.easing.min.js"></script>
    <script src="js/scrolling-nav.js"></script>
    
    <!-- Wizard JavaScript -->
	<script src="js/jquery.bootstrap.wizard.min.js"></script>    
	<script src="js/prettify.js"></script>	
	
	<!-- Widget JavaScript -->
	<script src="js/jasny-bootstrap.min.js"></script>
	
	<script>
	$(document).ready(function() {
	  	$('#rootwizard').bootstrapWizard({'tabClass': 'nav nav-pills'});	
		window.prettyPrint && prettyPrint()
	});	
	
	$(function () {
	  $('[data-toggle="popover"]').popover()
	});

	</script>	

</body>

</html>
